presto-spi/src/main/java/com/facebook/presto/spi/block/FixedWidthBlock.java

```diff
@@ -73,21 +74,13 @@ public int getPositionCount()
     @Override
     public int getSizeInBytes()
     {
-        long size = getRawSlice().length() + valueIsNull.length();
-        if (size > Integer.MAX_VALUE) {
-            return Integer.MAX_VALUE;
-        }
-        return (int) size;
+        return intSaturatedCast(getRawSlice().length() + valueIsNull.length());
     }
 
     @Override
     public int getRetainedSizeInBytes()
     {
-        long size = INSTANCE_SIZE + getRawSlice().getRetainedSize() + valueIsNull.getRetainedSize();
-        if (size > Integer.MAX_VALUE) {
-            return Integer.MAX_VALUE;
-        }
-        return (int) size;
+        return intSaturatedCast(INSTANCE_SIZE + getRawSlice().getRetainedSize() + valueIsNull.getRetainedSize());
```

Comment:
shouldn't we need to cache this value given that we found it will be used in a tight loop?